p2p/peer.go

```diff
@@ -490,35 +748,76 @@ func (s *peerStore) List() ([]*peerInfo, error) {
 	return peers, nil
 }
 
-// peerInfo contains peer information stored in a peerStore.
+// Ranked returns a list of peers ordered by score (better peers first).
+// Peers with equal scores are returned in an arbitrary order.
 //
-// FIXME: This should be renamed peer or something else once the old peer is
-// removed.
-type peerInfo struct {
-	ID        NodeID
-	Addresses []PeerAddress
+// This is used to determine which peers to connect to and which peers to evict
+// in order to make room for better peers.
+//
+// FIXME: For now, we simply generate the list on every call, but this can get
+// expensive since it's called fairly frequently. We may want to either cache
+// this, or store peers in a data structure that maintains order (e.g. a heap or
+// ordered map).
+func (s *peerStore) Ranked() ([]*peerInfo, error) {
+	peers, err := s.List()
+	if err != nil {
+		return nil, err
+	}
+	sort.Slice(peers, func(i, j int) bool {
+		// FIXME: If necessary, consider precomputing scores before sorting,
+		// to reduce the number of Score() calls.
+		return peers[i].Score() < peers[j].Score()
+	})
```

Comment:
It seems to me from looking at the example here: https://golang.org/pkg/sort/#Slice, that this will sort in ascending order but what we want is descending order right i.e. `(better peers first)`

Reply:
Ah, of course, thanks!